### PR TITLE
JSON-ize the CLI output

### PR DESCRIPTION
**Pull request by mhchia**
### What was wrong?
Fix https://github.com/ethresearch/sharding-p2p-poc/issues/94

### How was it fixed?
If a command succeeds, output the result directly.
e.g. 
```
$ sharding-p2p-poc -client getsubshard
[1,2,3,4,5]
```
If a command fails, output the error message, and set the return code as non-zero.
e.g.
```
$ sharding-p2p-poc -client broadcastcollation 1 1 1 100 0
Failed to broadcast 1 collations of size 100 in period 0 in shard 1, err: rpc error: code = Unknown desc = Failed to broadcast collation, err: broadcasting to a not subscribed shard
```

#### TODOs
- [ ] See if it fits the needs of Whiteblock team @araskachoi . If we do need status code, I can add it.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by mhchia** [comment]
TODO
- [ ] still need to modify `doXXX`

File: cli.go [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+	}
+	defer conn.Close()
+	client := pbrpc.NewPocClient(conn)
+	listPeerReq := &pbrpc.RPCListPeerRequest{}
+	logger.Debugf("rpcclient:ListPeer: sending=%v", listPeerReq)
+	res, err := client.ListPeer(context.Background(), listPeerReq)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to call RPC ListPeer at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
```
Comment (NIC619) [inline cli.go]: No `logger.Debugf("rpcclient:ListPeer: result=%v", res)`?

File: cli.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+	peers := &PeerResponse{
+		Peers: res.Peers.Peers,
```
Comment (NIC619) [inline cli.go]: `res.Peers` seems fine for the list of peers returned

File: cli.go [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+	client := pbrpc.NewPocClient(conn)
+	listTopicPeerReq := &pbrpc.RPCListTopicPeerRequest{
+		Topics: topics,
+	}
+	logger.Debugf("rpcclient:ListTopicPeer: sending=%v", listTopicPeerReq)
+	res, err := client.ListTopicPeer(context.Background(), listTopicPeerReq)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to call RPC ListTopicPeer at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
```
Comment (NIC619) [inline cli.go]: Ditto: no `logger.Debugf(…)`?

File: cli.go [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+type TopicPeerResponse struct {
+	TopicPeers map[string]([]string)
+}
+
+var emptyResponse interface{}
+
+func marshalToJSONString(data interface{}) string {
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to marshal object: %v", err)
```
Comment (NIC619) [inline cli.go]: Nitpick: `fmt.Sprintf("Failed to marshal object: %v, err: %v", data, err)`

File: cli.go [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+	dataStr := string(dataBytes)
+	return dataStr
+}
+
+func emitCLIFailure(msg string) {
+	resp := &CLIFailure{
+		Status:  false,
+		Message: msg,
+	}
+	respStr := marshalToJSONString(resp)
```
Comment (NIC619) [inline cli.go]: Will this results in a `marshalToJSONString` -> `emitCLIFailure` -> `marshalToJSONString` loop?

Comment (mhchia) [reply cli.go]: Yes, you're right. It will cause a recursion. Will fix it.

File: cli.go (hunk 2 above)
Comment (mhchia) [reply cli.go]: `res.Peers.Peers` is the `[]string`. `[]string` is needed for JSON marshaling

**Comment by mhchia** [comment]
cc @NIC619  @ChihChengLiang

**Comment by araskachoi** [comment]
@mhchia For the status code, could you return any non-0 value for any errors.

**Comment by mhchia** [comment]
@araskachoi  Yes, we have done it. Any error leads to a non-zero return value of the program

**Comment by araskachoi** [comment]
@mhchia awesome, thanks for the clarification!

**Comment by mhchia** [comment]
No problem:) I think the RPCs and output should be ready. Please give us any feedback if there is still any issue.

**Comment by araskachoi** [comment]
Awesome, I will pull and test it out and see if it is good! Also, I think it would be a valuable metric to add into the json, the timestamp for when any action has been successfully completed. This way we can measure the time it takes for something to happen (especially when we start testing with network conditions and large scale tests). Please let me know your thoughts!

**Pull request merged**